nvdaHelper/local/UIAEventLimiter/utils.h

```diff
@@ -0,0 +1,33 @@
+/*
+This file is a part of the NVDA project.
+URL: http://github.com/nvaccess/nvda/
+Copyright 2023 NV Access Limited.
+	This program is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License version 2.0, as published by
+	the Free Software Foundation.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+This license can be found at:
+http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+*/
+
+#include <concepts>
+#include <vector>
+#include <algorithm>
+#include <ranges>
+#include <variant>
+#include <uiautomation.h>
+#include <comutil.h>
+
+std::vector<int> SafeArrayToVector(SAFEARRAY* pSafeArray);
+
+std::vector<int> getRuntimeIDFromElement(IUIAutomationElement* pElement);
+
+template <typename T, typename V, std::size_t... indexes>
```

Comment:
I think I understand what supports_alternative  is for, but given how cryptic some of this stuff appears at first glance, please add a brief comment explaining what this does.